api/lib/domain/services/check-sco-account-recovery-service.js

```diff
@@ -1,25 +1,89 @@
-const { MultipleSchoolingRegistrationsWithDifferentNationalStudentIdError } = require('../errors');
+const { MultipleSchoolingRegistrationsWithDifferentNationalStudentIdError, UserNotFoundError } = require('../errors');
 const _ = require('lodash');
 
 async function retrieveSchoolingRegistration({
   studentInformation,
   schoolingRegistrationRepository,
   userRepository,
+  userReconciliationService,
 }) {
 
-  const { id, userId, firstName, lastName, organizationId } = await schoolingRegistrationRepository.getSchoolingRegistrationInformation({
-    ...studentInformation,
-    nationalStudentId: studentInformation.ineIna,
+  const lastSchoolingRegistration = await _getLastMatchingSchoolingRegistration({
+    ineIna: studentInformation.ineIna,
+    birthdate: studentInformation.birthdate,
+    schoolingRegistrationRepository,
   });
+
+  const userId = await _getUserIdByMatchingStudentInformationWithSchoolingRegistration({
+    studentInformation,
+    lastSchoolingRegistration,
+    userReconciliationService,
+  });
+
+  await _checkIfThereAreMultipleUserForTheSameAccount({ userId, schoolingRegistrationRepository });
+
+  const { username, email } = await userRepository.get(userId);
+
+  return {
+    id: lastSchoolingRegistration.id,
+    userId: lastSchoolingRegistration.userId,
+    firstName: lastSchoolingRegistration.firstName,
+    lastName: lastSchoolingRegistration.lastName,
+    username,
+    organizationId: lastSchoolingRegistration.organizationId,
+    email,
+  };
+}
+
+async function _getLastMatchingSchoolingRegistration({
+  ineIna,
+  birthdate,
+  schoolingRegistrationRepository,
+}) {
+  const schoolingRegistrationsMatchingIneAndBirthdate = await schoolingRegistrationRepository
+    .findByNationalStudentIdAndBirthdate({
+      nationalStudentId: ineIna,
+      birthdate: birthdate,
+    });
+
+  const reconciledschoolingRegistrationsMatchingIneAndBirthdate = schoolingRegistrationsMatchingIneAndBirthdate
+    .filter((schoolingRegistration) => !_.isNil(schoolingRegistration.userId));
+
+  if (reconciledschoolingRegistrationsMatchingIneAndBirthdate.length === 0) {
+    throw new UserNotFoundError();
+  }
+
+  const lastSchoolingRegistration = _(schoolingRegistrationsMatchingIneAndBirthdate)
+    .orderBy('updatedAt')
+    .reverse()
+    .first();
```

Comment:
Pour éviter une étape superflue:
```suggestion
  const lastSchoolingRegistration = _(schoolingRegistrationsMatchingIneAndBirthdate)
    .orderBy('updatedAt')
    .last();
```

Reply:
Bien vu.. mais le code sera supprimé par https://github.com/1024pix/pix/pull/3230#discussion_r677268526